### DISCUSSION TITLE
Tree shaking with Webpack / Typescript

### DISCUSSION THREAD

**Discussion by williamsdyyz** [panva/jose#553] [Q&A]
First of all: Thank you for this library! It's made my life infinitely easier

I have a React site. I'm using Jose to decode and validate a JWT and it works great.

The only problem is the bundle size which according to webpack-bundle-analyzer weighs in at 131.0k KB. 

Tree shaking is known to be working in this project so I don't think it's a general problem. Seems to be specific to this library.

I'm importing importSPKI, compactVerify and the type KeyLike, but the bundle analysis shows a lot of files that I wouldn't expect. 

Does this look correct? If not do you have any thoughts on how to track down the problem?

tsconfig compilerOptions.module is "ES2020" and I'm importing like this (tried import * as... with same result)
```
import { importSPKI, compactVerify } from "jose";
import type { KeyLike } from "jose";
```

webpack-bundle-analyzer output:
![Screenshot 2023-07-19 111943](https://github.com/panva/jose/assets/35264112/fbb41ffc-cc41-44ce-9760-0486a8200548)

**Comment by panva** [comment]
Have you entertained the possibility of your bundle analyzer not taking treeshaking into consideration?

**Reply by williamsdyyz** [reply]
It's aware of tree shaking - big chunks of other libraries are not in the report.

**Reply by panva** [reply]
As you can see the included files breakdown it's not treeshaking correctly and includes the whole library ;)

**Reply by williamsdyyz** [reply]
That's why I'm asking the question :)  It's working for every library except this one and I haven't been able to figure out why. I thought since  Jose was specifically documented as being tree shakable someone might have some deeper understanding of what could go wrong

**Comment by remorses** [comment]
Currently Jose takes about 100ms to be required in an AWS lambda function

Being able to only require the functions you need could significantly decrease cold starts for many projects:

![Screenshot 2023-08-18 at 17 05 02](https://github.com/supabase/supabase-js/assets/31321188/dadab937-8ba0-49b5-9e48-7c5ed873f430)


[require profile here](https://ucarecdn.com/a21718eb-b5bf-489b-80bf-2c899970c9da/)
